### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by diehualong** [JiwanChung/esper#1]
Can you tell me the GPU used and the training time? Thank you for your time.

**Comment by JiwanChung** [comment]
1. We used a single Nvidia RTX TITAN (24GB) per experiment. We early-stopped each run after predefined epochs, which typically took 3-4 days. The major bottleneck was the inference time required for RL exploration.
2. With that said, I think as of 2023 using a better GPU (possibly with <48GB VRAM) and multi-node training would scale the running time efficiency well. Also, you can use weight quantization techniques and LORAs for lighter inference & optimization.

**Comment by diehualong** [comment]
Hello. I am using text/code/train_generator.py and coco caption to fine tune gpt2, the commands and the image of the loss function on the tensorboard are shown below. I would like to ask if there is something wrong with the shape of this loss function?
![image](https://github.com/JiwanChung/esper/assets/69414776/7bf262fe-0af0-402d-9b3a-52510fb1d6ac)
![image](https://github.com/JiwanChung/esper/assets/69414776/51f37f49-4673-4cbc-b00b-e4406d5b1679)

**Comment by JiwanChung** [comment]
Hi! First of all, thank you for your attention to our project.

The answers are twofold:
1. I suggest you use the default config file `https://github.com/JiwanChung/esper/blob/master/train/code/configs/esper.yml` to reproduce our results as described in ReadMe. Without it, the training script will use the default argument values provided in `https://github.com/JiwanChung/esper/blob/master/train/code/arguments.py`, which is not ideal.
2. Looking at your loss function, I think that the objective is not being optimized properly. Please modify your hyperparameters as in the above suggestion.

**Comment by JiwanChung** [comment]
I'm closing this issue assuming my response was helpful. Please feel free to reopen it should you have any further inquiries :)
